### DISCUSSION TITLE
Make workflow run after a list of workflows has finished successfuly

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#46487] [Actions] [answered]
### Select Topic Area

Question

### Body

Assuming I have a list of workflow files such as `workflow1.yaml`, `workflow2.yaml`, `workflow3.yaml`.

Is there a way to configure `workflow3.yaml` to run if and only if `workflow1.yaml` and `workflow2.yaml` have finished with **success**?

**Comment by pantelis-karamolegkos** [comment]
Thanks. Are you sure the `needs` keyword can be used for `workflows` dependencies? Haven't seen that before. Have you come across any relevant documentation?

**Comment by swathida-byte** [comment, marked as answer]
According to[ Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) , You can use 
on:
  workflow_run:
      types:
         - completed

**Comment by pantelis-karamolegkos** [comment]
I think that is the correct approach. It has some limitations though such as:

- you need a workaround (check each job of the last workflow down the chain if the previous workflow was successful)
- only **one** in the list of previous workflows needs to be completed for the last one to run
